File: 1.py
from typing import List
class Solution:
    def twoSum(nums: List[int], target: int) -> List[int]:
        dictionary = dict()
        for idx, value in enumerate(nums):
            to_find = target - value
            if to_find in dictionary:
                return [idx, dictionary[to_find]]
            dictionary[value] = idx
        return []

File: 1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] answer = new int[2];
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++) {
            int toFind = target - nums[i];
            if(map.containsKey(toFind)) {
                answer[0] = i;
                answer[1] = map.get(toFind);
                break;
            }
            map.put(nums[i], i);
        }
        return answer;
    }
}